controller_manager:
  ros__parameters:
    update_rate: 50  # Hz

    arm_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

arm_trajectory_controller:
  ros__parameters:
    joints:
      - joint0
      - joint1
      - joint2
    
    command_interfaces:
      - position
    
    state_interfaces:
      - position
      
    state_publish_rate: 50.0
    action_monitor_rate: 20.0

    allow_partial_joints_goal: false
    constraints:
      stopped_velocity_tolerance: 0.01
      goal_time: 0.0